lib/get-esm-exports.js

```diff
@@ -14,81 +17,187 @@ function warn (txt) {
   process.emitWarning(txt, 'get-esm-exports')
 }
 
-function getEsmExports (moduleStr) {
-  const exportedNames = new Set()
-  const tree = parser.parse(moduleStr, acornOpts)
-  for (const node of tree.body) {
-    if (!node.type.startsWith('Export')) continue
-    switch (node.type) {
-      case 'ExportNamedDeclaration':
-        if (node.declaration) {
-          parseDeclaration(node, exportedNames)
-        } else {
-          parseSpecifiers(node, exportedNames)
+function getEsmExports(moduleStr, generate=false, url=undefined) {
+  const exportSpecifierNames = new Set()
+  const exportAlias = {}
+  let ast
+  
+  // if it's a typescript file, we need to parse it with recasts typescript parser
+  if (url && TS_EXTENSION_RE.test(url)) {
+    ast = recast.parse(moduleStr, {parser: require("recast/parsers/typescript")})
+  } else {
+    ast = recast.parse(moduleStr, {parser: {
+      parse(source) {
+        return parser.parse(source, acornOpts)
+      }
+    }})
+  }
+
+  recast.visit(ast, {
+    visitExportNamedDeclaration(path) {
+      const node = path.node
+  
+      if (node.declaration) {
+        parseDeclaration(node.declaration, exportAlias)
+      } else {
+        parseSpecifiers(node.specifiers, exportAlias, exportSpecifierNames)
+      }
+  
+      this.traverse(path)
+    },
+  
+    visitExportDefaultDeclaration(path) {
+      const node = path.node
+      const iitmRenamedExport = 'iitmRenamedExport'
+  
+      if ((node.declaration.type === 'ObjectExpression' || node.declaration.type === 'ArrayExpression' || node.declaration.type === 'Literal') && generate) {
```

Comment:
Avoid doing the object access 3 times like this. Consider stashing `node.declaration.type` in a variable, or doing something like: `['ObjectExpression', 'ArrayExpression', 'Literal'].includes(node.declaration.type)`